CONTRIBUTING.md

```diff
@@ -151,3 +151,44 @@ The OTLP proto dependency version is defined [here](proto/build.gradle). To bump
 2. Download the zip source code archive
 3. Run `shasum -a 256 ~/path/to/downloaded.zip` to compute its checksum
 4. Update `protoVersion` and `protoChecksum` in the build file with the new version and checksum
+
+### Composing builds
+
+Beware that this section is only meant for developers of opentelemetry-java, or closely related projects.
+The steps described here could change at any time and what you do for one version (commit) may break
+with the next one already.
+
+Gradle provides a feature called ["composite builds"](https://docs.gradle.org/current/userguide/composite_builds.html)
+that allows to replace some normally externally provided dependencies with a project that is built
+(included) in the same Gradle invocation. This can be useful to quickly test a new feature or bug fix you are
+developing in opentelmetry-java with the examples or the app or instrumentation library where you
+need the feature or run into the bug. Unfortunately, opentelemetry-java does not work out of the box
+with this feature because Gradle is unable to map the project names to the customized artifact
+coordinates (see e.g. [gradle/gradle#18291](https://github.com/gradle/gradle/issues/18291)
+and related issues. However, gradle supports manually declaring the mapping between ("substitution of")
+artifact coordinates and project names. To ease this tedious task, opentelemetry-java provides a
+gradle task `:generateBuildSubstitutions` that generates a code snippet with these substitutions in
+kts (Kotlin Script) format.
+
+Example usage could be as follows:
+
+1. Run `./gradlew generateBuildSubstitutions`
+2. Two files named `build/substitutions.gradle.kts` are generated in the bom and bom-alpha project's
+   directory, containing substitutions for the stable and alpha projects respectively.
+3. Copy & paste the content of these files to a new `settings.gradle.kts` or the one where you want
```

Comment:
This is still cumbersome. If anyone knows a better format I could generate that could be used in a more straight-forward way, that would be appreciated.